[GEOT-5627] Fixed create temp file

ugomos opened the pull request.

simboss commented on modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/ImageMosaicReaderTest.java:
```diff
@@ -2047,7 +2047,7 @@ public void testRequestInAreaWithNoGranulesBecomesTransparent() throws Exception
         GridCoverage2D coverage = reader.read(new GeneralParameterValue[] {ggp, transparent});
         assertNotNull(coverage);
-        new GeoTiffWriter(new File("/tmp/test.tiff")).write(coverage, null);
+        new GeoTiffWriter(TestData.temp(this, "test.tiff")).write(coverage, null);
```
I would add a call to dispose at the end.

simboss commented:
Do we also need to backport?

Please, check if this applies to older versions.

ugomos commented:
I updated this PR adding the dispose of the ImageMosaicReader.
Yes, we need also to backport, the same problem is at least in branch 16.x and 15.x.

ugomos commented:
At last, the line "new GeoTiffWriter(new File("/tmp/test.tiff"))" had no reason to exist, it is now deleted. Added the dispose of the ImageMosaicReader object.
